components/select_team/select_team.tsx

```diff
@@ -29,35 +28,45 @@ import FormattedMarkdownMessage from 'components/formatted_markdown_message';
 
 import InfiniteScroll from '../common/infinite_scroll.jsx';
 
-import SelectTeamItem from './components/select_team_item.jsx';
+import SelectTeamItem from './components/select_team_item';
 
 export const TEAMS_PER_PAGE = 30;
 const TEAM_MEMBERSHIP_DENIAL_ERROR_ID = 'api.team.add_members.user_denied';
 
-export default class SelectTeam extends React.PureComponent {
-    static propTypes = {
-        currentUserId: PropTypes.string.isRequired,
-        currentUserRoles: PropTypes.string,
-        currentUserIsGuest: PropTypes.bool,
-        customDescriptionText: PropTypes.string,
-        isMemberOfTeam: PropTypes.bool.isRequired,
-        listableTeams: PropTypes.array,
-        siteName: PropTypes.string,
-        canCreateTeams: PropTypes.bool.isRequired,
-        canManageSystem: PropTypes.bool.isRequired,
-        canJoinPublicTeams: PropTypes.bool.isRequired,
-        canJoinPrivateTeams: PropTypes.bool.isRequired,
-        history: PropTypes.object,
-        siteURL: PropTypes.string,
-        actions: PropTypes.shape({
-            getTeams: PropTypes.func.isRequired,
-            loadRolesIfNeeded: PropTypes.func.isRequired,
-            addUserToTeam: PropTypes.func.isRequired,
-        }).isRequired,
-        totalTeamsCount: PropTypes.number.isRequired,
-    };
+type Actions = {
+    getTeams: any,
+    loadRolesIfNeeded: any,
+    addUserToTeam: any,
+}
 
-    constructor(props) {
+type Props = {
+    currentUserId: string,
+    currentUserRoles?: string,
```

Comment:
Some of these like `currentUserRoles` can actually be made required since they're always passed in by `mapStateToProps`. Maybe you could make those required to avoid some ugly null checks below?

Reply:
Thanks. Made this change.